### PR TITLE
SAK-49436 Site Import: Import is 3x slower because how Rubrics are co…

### PR DESCRIPTION
**Pull request by mpellicer**
…pied in Tests & Quizzes

### CODE REVIEW

File: rubrics/api/src/main/java/org/sakaiproject/rubrics/api/RubricsService.java [hunk 1]
```diff
@@ -146,4 +146,13 @@ byte[] createPdf(String siteId, Long rubricId, String toolId, String itemId, Str
 
     void deleteSiteRubrics(String siteId);
 
+    /**
+     * Find the associations of a rubric to a tool
+     *
+     * @param rubricId the rubric id
+     * @param toolId the tool id, something like "sakai.assignment"
+     * @return a List of ToolItemRubricAssociation containing the associations of a Rubric to a tool. Example: All the rubric associations to assessmnets.
```
Comment (mpellicer) [inline rubrics/api/src/main/java/org/sakaiproject/rubrics/api/RubricsService.java:154]:
```suggestion
     * @return a List of ToolItemRubricAssociation containing the associations of a Rubric to a tool. Example: All the rubric associations to assessments.
```

File: rubrics/api/src/main/java/org/sakaiproject/rubrics/api/RubricsService.java [hunk 2]
```diff
@@ -146,4 +146,13 @@ byte[] createPdf(String siteId, Long rubricId, String toolId, String itemId, Str
...
     void deleteSiteRubrics(String siteId);
 
+    /**
+     * Find the associations of a rubric to a tool
+     *
+     * @param rubricId the rubric id
+     * @param toolId the tool id, something like "sakai.assignment"
+     * @return a List of ToolItemRubricAssociation containing the associations of a Rubric to a tool. Example: All the rubric associations to assessments.
+     */
+    List<ToolItemRubricAssociation> getRubricAssociationByRubricAndTool(Long rubricId, String toolId);
```
Comment (mpellicer) [inline rubrics/api/src/main/java/org/sakaiproject/rubrics/api/RubricsService.java:156]:
```suggestion
    List<ToolItemRubricAssociation> getRubricAssociationsByRubricAndTool(Long rubricId, String toolId);
```

File: rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/RubricsServiceImpl.java [hunk 3]
```diff
@@ -1620,4 +1620,9 @@ private Rubric updateRubricMaxPoints(Rubric rubric) {
         rubric.setMaxPoints(maxPoints);
         return rubric;
     }
+
+    public List<ToolItemRubricAssociation> getRubricAssociationByRubricAndTool(Long rubricId, String toolId) {
```
Comment (mpellicer) [inline rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/RubricsServiceImpl.java:1624]:
```suggestion
    public List<ToolItemRubricAssociation> getRubricAssociationsByRubricAndTool(Long rubricId, String toolId) {
```

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentFacadeQueries.java [hunk 4]
```diff
@@ -1599,7 +1600,19 @@ public void copyAllAssessments(String fromContext, String toContext, Map<String,
 		List<AssessmentData> list = getAllActiveAssessmentsByAgent(fromContext);
 		List<AssessmentData> newList = new ArrayList<>();
 		Map<AssessmentData, String> assessmentMap = new HashMap<>();
+
 		RubricsService rubricsService = (RubricsService) SpringBeanLocator.getInstance().getBean("org.sakaiproject.rubrics.api.RubricsService");
+		List<RubricTransferBean> siteRubricList = rubricsService.getRubricsForSite(fromContext);
+		List<String> rubricsInUseAssociationList = new ArrayList<>();
+		for (RubricTransferBean rubric : siteRubricList) {
+			Long rubricId = rubric.getId();
+			List<ToolItemRubricAssociation> rubricAssociationList = rubricsService.getRubricAssociationByRubricAndTool(rubricId, RubricsConstants.RBCS_TOOL_SAMIGO);
```
Comment (mpellicer) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentFacadeQueries.java:1609]:
```suggestion
			List<ToolItemRubricAssociation> rubricAssociationList = rubricsService.getRubricAssociationsByRubricAndTool(rubricId, RubricsConstants.RBCS_TOOL_SAMIGO);
```

**Review by bgarciaentornos — APPROVED** [review]

**Comment by mpellicer** [comment]
Thanks for the review

**Pull request merged**